File: c#/classes/inheritance/AbstractFunctions.cs
abstract class Value { // Must be abstract class to have abstract method.
  public double Valuation;
  public string Label;

  public Value(double valuation, string label = "") {
    Valuation = valuation;
    Label = label;
  }

  public abstract void Show();
}

class Datum: Value {
  public string Description;

  public Datum(double valuation, string label, string description = "") : base(valuation, label) {
    Description = description;
  }

  public override void Show() {
    System.Console.WriteLine("Value: " + Valuation
      + "\nLabel: " + Label
      + "\nDescription: " + Description
    );
  }
}

class Content: Datum {
  public Content(double value, string label, string description) : base(value, label, description) {
  }
};

class AbstractFunctions {
  public static void Main(string[] args) {
    // Value temperature = new Value(24, "Health for humans temperature in degree Celsius"); // One cannot instantiate.

    Content page = new Content(666, "Page of Harry Potter book", "The satanistic ritual hidden in the book for kids. Oh noes!");
    page.Show();
  }
}

File: java/classes/abstract_classes/AbstractFunctions.java
abstract class Value { // Must be abstract class or interface to have abstract method.
  public double value;
  public String label;

  public Value(double value_value, String value_label) {
    value = value_value;
    label = value_label;
  }

  public abstract void show();
}

class Datum extends Value {
  public String description;

  public Datum(double value_value, String value_label, String value_description) {
    super(value_value, value_label);
    description = value_description;
  }

  public void show() {
    System.out.println("Value: " + value
      + "\nLabel: " + label
      + "\nDescription: " + description
    );
  };
}

class Content extends Datum {
  public Content(double value, String label, String description) {
    super(value, label, description);
  }
};

class AbstractFunctions {
  public static void main(String[] args) {
    // Value temperature = new Value(24, "Health for humans temperature in degree Celsius"); // One cannot instantiate.

    Content page = new Content(666, "Page of Harry Potter book", "The satanistic ritual hidden in the book for kids. Oh noes!");
    page.show();
  }
}
